lib/buildkite/test_collector.rb

```diff
@@ -77,6 +77,18 @@ def self.annotate(content)
       tracer&.leave
     end
 
+    # Set a key=value tag on the current test execution.
+    def self.tag_execution(key, value)
+      tags = Thread.current[:_buildkite_tags]
+      raise "_buildkite_tags not available" unless tags
+
+      unless key.is_a?(String) && value.is_a?(String)
```

Comment:
Should we consider supporting String like classes like numbers and booleans?

Should we be more strict about sanitisation or should we handle that server side?

Reply:
Unsure, so I started strict and we can expand it to be more permissive without breaking anything for anyone.
(Can't do the other direction)